src/puya/awst/serialize.py

```diff
@@ -86,4 +88,7 @@ def awst_to_json(awst: nodes.AWST) -> str:
 
 
 def awst_from_json(json: str) -> nodes.AWST:
-    return _get_converter().loads(json, nodes.AWST)  # type: ignore[type-abstract]
+    try:
+        return _get_converter().loads(json, nodes.AWST)  # type: ignore[type-abstract]
+    except (ClassValidationError, IterableValidationError, BaseException) as err:
+        raise InternalError("Deserialization error: \n" + "\n".join(transform_error(err))) from err
```

Comment:
But also, since this sort of error details would only really be useful for anyone working on the compiler(s), I'm wondering whether putting all this data into the exception message is the best experience in case anyone encounters it? It might be better to keep it succinct, and just log the details (ie results of `transform_error`) as debug messages.

Reply:
Moved details of error into a `logger.debug` and kept the raised error message brief